app/dao/fact_notification_status_dao.py

```diff
@@ -440,3 +448,67 @@ def get_total_sent_notifications_for_day_and_type(day, notification_type):
     ).scalar()
 
     return result or 0
+
+
+def fetch_monthly_notification_statuses_per_service(start_date, end_date):
+    return db.session.query(
+        func.date_trunc('month', FactNotificationStatus.bst_date).cast(Date).label('date_created'),
+        Service.name.label('service_name'),
+        Service.id.label('service_id'),
+        FactNotificationStatus.notification_type,
+        func.sum(case(
+            [
+                (FactNotificationStatus.notification_status == NOTIFICATION_SENDING,
+                 FactNotificationStatus.notification_count)
+            ],
+            else_=0)).label('count_sending'),
+        func.sum(case(
+            [
+                (FactNotificationStatus.notification_status == NOTIFICATION_DELIVERED,
+                 FactNotificationStatus.notification_count)
+            ],
+            else_=0)).label('count_delivered'),
+        func.sum(case(
+            [
+                (FactNotificationStatus.notification_status.in_([NOTIFICATION_TECHNICAL_FAILURE, NOTIFICATION_FAILED]),
+                 FactNotificationStatus.notification_count)
+            ],
+            else_=0)).label('count_technical_failure'),
+        func.sum(case(
+            [
+                (FactNotificationStatus.notification_status == NOTIFICATION_TEMPORARY_FAILURE,
+                 FactNotificationStatus.notification_count)
+            ],
+            else_=0)).label('count_temporary_failure'),
+        func.sum(case(
+            [
+                (FactNotificationStatus.notification_status == NOTIFICATION_PERMANENT_FAILURE,
+                 FactNotificationStatus.notification_count)
+            ],
+            else_=0)).label('count_permanent_failure'),
+        func.sum(case(
+            [
+                (FactNotificationStatus.notification_status == NOTIFICATION_SENT,
+                 FactNotificationStatus.notification_count)
+            ],
+            else_=0)).label('count_sent'),
+    ).join(
+        Service, FactNotificationStatus.service_id == Service.id
+    ).filter(
+        FactNotificationStatus.notification_status != NOTIFICATION_CREATED,
```

Comment:
why we don't count created notifications? Is it because they can be still cancelled?

Reply:
I'm not sure what the report gets used for - just trying to copy the query the engagement team already gets

Reply:
The engagement team are not interested in the created notifications. To be fair this report is usually asked for after the end of the month, so there should really be anything still in created.